numpy/lib/tests/test_index_tricks.py

```diff
@@ -262,6 +262,24 @@ def test_accepts_npfloating(self):
         assert_(grid32.dtype == np.float64)
         assert_array_almost_equal(grid64, grid32)
 
+    def test_accepts_longdouble(self):
+        # regression tests for #16945
+        grid64 = mgrid[0.1:0.33:0.1,]
+        grid128 = mgrid[np.longdouble(0.1):np.longdouble(0.33):np.longdouble(0.1),]
+        assert_(grid128.dtype == np.longdouble)
+        assert_array_almost_equal(grid64, grid128)
+
+        grid128c_a = mgrid[0:np.longdouble(1):3.4j]
+        grid128c_b = mgrid[0:np.longdouble(1):3.4j,]
+        assert_(grid128c_a.dtype == grid128c_b.dtype == np.longdouble)
+        assert_array_equal(grid128c_a, grid128c_b[0])
+
```

Comment:
Lets add one more of these:
```suggestion

        grid128c_a = mgrid[0:1:np.clongdouble(3.4j]
        grid128c_b = mgrid[0:1:np.clongdouble(3.4j),]
        assert_(grid128c_a.dtype == grid128c_b.dtype == np.longdouble)
        assert_array_equal(grid128c_a, grid128c_b[0])
```

Reply:
This was a good call actually. `grid128c_a` had the `float64` dtype while `grid128c_b` was of `float128` 
(which one would expect as `type(abs(np.complex256(...))) is np.float128`).

Anyway, it's been fixed as of https://github.com/numpy/numpy/pull/16971/commits/c093dcb90d5c5de1e7d8af0ea2681567316cf0e0 and the tests have been updated in https://github.com/numpy/numpy/pull/16971/commits/3a49c7c750fddc2093700d8d844155be86c0f302.

Reply:
I think this was the bug I was trying to catch above.